File: HealthcareSystem.Application/Requests/ScheduleRequests.cs
namespace HealthcareSystem.Application.Requests;

public static class ScheduleRequests {
    public record GetSchedulesRequest(
        Guid? DoctorId, int? PageIndex, int? PageSize,
        DateTimeOffset? StartTime, DateTimeOffset? EndTime
    );

    public record UpdateScheduleRequest(bool IsAvailable);

    public record CreateScheduleRequest(
        Guid DoctorId, DateTimeOffset StartTime, DateTimeOffset EndTime
    );
}

File: HealthcareSystem.Client/src/app/interfaces/requests/scheduleRequests.ts
export interface GetSchedulesRequest {
  doctorId?: string;
  pageIndex?: number;
  pageSize?: number;
  startTime?: Date;
  endTime?: Date;
}

export interface UpdateScheduleRequest {
  isAvailable: boolean;
}

export interface CreateScheduleRequest {
  doctorId: string;
  startTime?: Date;
  endTime?: Date;
}
